### ISSUE TITLE
Support for Babel 5.0.0

### ISSUE THREAD

**Issue by skevy** [babel/babel-eslint#61]
Just gonna leave this here so I can track it. I'm sure it's already in the works. I started attempting last night, but to be honest I wasn't sure how to attack the eslint cleanup that happens in https://github.com/babel/babel-eslint/blob/master/acorn-to-esprima.js#L52

I feel like it's probably relatively straightforward - just wasn't sure how to do it.

**Comment by sebmck** [comment]
I'm sitting on some commits to do this actually, will hopefully get the
chance to finish it once the work from 5.0.0 settles.

On Friday, 3 April 2015, Adam Miskiewicz <notifications@github.com> wrote:

> Just gonna leave this here so I can track it. I'm sure it's already in the
> works. I started attempting last night, but to be honest I wasn't sure how
> to attack the eslint cleanup that happens in
> https://github.com/babel/babel-eslint/blob/master/acorn-to-esprima.js#L52
>
> I feel like it's probably relatively straightforward - just wasn't sure
> how to do it.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/babel/babel-eslint/issues/61>.
>


-- 
Sebastian McKenzie

**Comment by skevy** [comment]
Figured as much. That's why I didn't try too much harder...I guessed you were already working on it. Thanks so much...and I'm loving Babel 5 (decorators for the win!).

**Comment by deepsweet** [comment]
I have a bunch of

```
  1:0  error  'import' and 'export' may appear only with 'sourceType: module'
```

only in one of my repos (and I have no idea why exactly in this one), downgrading to `babel*@4` works. hope this may help to collect and fix issues with `babel-eslint` + `babel@5`.

**Comment by bytasv** [comment]
Is there a fast fix for:

```
error  Unexpected character '@'
```

Which can't not be removed even by using `// eslint-disable-line` on that line..

Here's the code:
```es6
class Cat extends Animal {

  @isSiamese // eslint-disable-line
  @hasParam('tail') // eslint-disable-line
  woof() {
    // ...
  }
}
```

What are my options in this case that does not include **disabling linter** or **removing decorators**?

**Comment by sebmck** [comment]
@bytasv 

> Is there a fast fix for:

No.

> What are my options in this case that does not include disabling linter or removing decorators?

Wait until I get time to update `babel-eslint` for Babel 5.0.0

**Comment by sebmck** [comment]
Extremely sorry for the delay. I've published `babel-eslint` 3.0.0 to npm that uses `babel` 5.0.0.

**Comment by sebastienbarre** [comment]
@sebmck Ha, I'm afraid this is breaking the ES7+ Property Initializers check. Upgrading to 3.0.0 now triggers errors that weren't triggered before by 2.0.2 (using babel 5.1.8).
```
src/js/clients/ColorClient.js
  13:9  error  "STATUS" is not defined  no-undef
```
in
```
class ColorClient {
  static STATUS = {
    OK: 0,
    LOADING_COLORS: 1,
    LOADING_COLORS_COMPLETED: 2,
    LOADING_COLORS_FAILED: 3
  }
[...]
```

**Comment by sebmck** [comment]
Should be fixed as of 3.0.1.

**Comment by sebastienbarre** [comment]
Confirmed, thanks for the quick fix.
